prometheus-to-sd/config/api_server_client.go

```diff
@@ -0,0 +1,68 @@
+package config
+
+import (
+	"errors"
+	"fmt"
+
+	core "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	labelForMonitoredPods = "prom-to-sd.googleapis.com=true"
+)
+
+// GetConfigsFromApiServer takes pod specifications from the Kubernetes API and maps them to source configs
+func GetConfigsFromApiServer(gceConfig *GceConfig, namespace string) ([]SourceConfig, error) {
+	podsApi, err := setupKubernetesPodsApi(namespace)
+	if err != nil {
+		return nil, err
+	}
+	podResponse, err := podsApi.List(v1.ListOptions{
+		FieldSelector: fmt.Sprintf("spec.nodeName=%s", gceConfig.Instance),
+		LabelSelector: labelForMonitoredPods,
+	})
+	if err != nil {
+		return nil, err
+	}
+	pods := podResponse.Items
+	if len(pods) < 0 {
+		return nil, errors.New("no pods for monitoring were found")
+	}
+	return getConfigsFromPods(pods)
+}
+
+func setupKubernetesPodsApi(namespace string) (corev1.PodInterface, error) {
```

Comment:
It will be more reusable to have function that returns k8s client, not particular api.